### PR TITLE
Fix dpi property

### PR DESCRIPTION
**Pull request by vladest**
The fix implemented parsing messages with properties updates from html/js world
Since QScriptEngine not capable to work in non-GUI thread its better to pass properties thru QVariant

Test plan:
- press Create button, create Web entiy
- on Properties tab scroll down to DPI settings. Change DPI.
- make sure DPI gets changed
- make sure other properties updates still works

check case https://highfidelity.fogbugz.com/f/cases/5182/Can-t-change-Resolution-DPI-on-a-web-entity

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1875/HighFidelity-Beta-PR11027-90e461cc152dfbb9a5b9b98e6c8db39f17c831e5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1875/HighFidelity-Beta-PR11027-90e461cc152dfbb9a5b9b98e6c8db39f17c831e5.dmg

File: libraries/display-plugins/src/display-plugins/DisplayPlugin.cpp [hunk 1]
```diff
@@ -25,9 +25,9 @@ DisplayPluginList getDisplayPlugins() {
     DisplayPlugin* PLUGIN_POOL[] = {
         new Basic2DWindowOpenGLDisplayPlugin(),
         new DebugHmdDisplayPlugin(),
-#ifdef DEBUG
-        new NullDisplayPlugin(),
-#endif
+//#ifdef DEBUG
```
Comment (ZappoMan) [inline libraries/display-plugins/src/display-plugins/DisplayPlugin.cpp]: don't comment out this code, please repair

File: libraries/entities/src/AnimationPropertyGroup.cpp [hunk 2]
```diff
@@ -76,9 +76,49 @@ void AnimationPropertyGroup::copyFromScriptValue(const QScriptValue& object, boo
         COPY_PROPERTY_FROM_QSCRIPTVALUE_GETTER(animationIsPlaying, bool, setRunning, getRunning);
         COPY_PROPERTY_FROM_QSCRIPTVALUE_GETTER(animationFrameIndex, float, setCurrentFrame, getCurrentFrame);
     }
+}
+
+void AnimationPropertyGroup::copyFromVariant(const QVariant &object, bool &_defaultSettings)
```
Comment (ZappoMan) [inline libraries/entities/src/AnimationPropertyGroup.cpp]: coding standard: brace placement

File: libraries/entities/src/EntityItemProperties.h [hunk 3]
```diff
@@ -304,8 +305,7 @@ class EntityItemProperties {
     bool getRenderInfoHasTransparent() const { return _renderInfoHasTransparent; }
     void setRenderInfoHasTransparent(bool value) { _renderInfoHasTransparent = value; }
 
-
-protected:
+    protected:
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItemProperties.h]: formatting - outdent this

File: script-archive/example/messages/messagesTestSend.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ var messageIndex = 1;
 
 var messageRate = 500;
 function sendMessage(){
-    print('SENDING MESSAGE')
+//    print('SENDING MESSAGE')
```
Comment (ZappoMan) [inline script-archive/example/messages/messagesTestSend.js]: accident?

**Comment by YuppyPlays** [comment]
Hi @vladest   
I can test if needed though I do not have access to view the fogbugz case, can assist with the info required :)

File: libraries/shared/src/RegisteredMetaTypes.cpp [hunk 5]
```diff
@@ -430,6 +430,19 @@ QVector<glm::vec3> qVectorVec3FromScriptValue(const QScriptValue& array){
     return newVector;
 }
 
+//QVector<glm::vec3> qVectorVec3FromVariant(const QVariant& array){
```
Comment (ZappoMan) [inline libraries/shared/src/RegisteredMetaTypes.cpp]: dead code? 

Why is this here but commented out?

File: libraries/entities/src/AnimationPropertyGroup.cpp (hunk 2 above)
Comment (jherico) [inline libraries/entities/src/AnimationPropertyGroup.cpp]: also, references should have the `&` next to the type, not the variable name, i.e. `const QVariant& object`.

File: libraries/ui/CMakeLists.txt [hunk 6]
```diff
@@ -1,6 +1,10 @@
 set(TARGET_NAME ui)
 setup_hifi_library(OpenGL Network Qml Quick Script WebChannel WebEngine WebSockets XmlPatterns)
-link_hifi_libraries(shared networking gl audio)
+link_hifi_libraries(shared networking gl audio entities)
+
+include_directories("${PROJECT_SOURCE_DIR}/../../libraries/entities/src")
```
Comment (jherico) [inline libraries/ui/CMakeLists.txt]: you can use `include_hifi_library_headers` here instead of a relative path...  ex:  `include_hifi_library_headers(entities)`

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 7]
```diff
@@ -61,6 +64,23 @@ void EntityScriptingInterface::resetActivityTracking() {
     _activityTracking.editedEntityCount = 0;
 }
 
+void EntityScriptingInterface::parseEntityMessage(const QVariant &message) {
+    if (message.isValid()) {
+        const QByteArray &messageba = message.toByteArray();
+        if (messageba.contains("id") &&
```
Comment (jherico) [inline libraries/entities/src/EntityScriptingInterface.cpp]: this logic is super dangerous.  This appears to get invoked from every message coming from the QML sent to the scripts, and intercepting them and passing them to this scripting interface on the basis of a few pattern matches is likely to eventually end up improperly treating some non-entity related message being interpreted as an edit command.  This isn't an acceptable approach.  We can expose the entity scripting interface to QML and allow QML objects to explicitly call edit commands when they need to, although we may need a new function so that the calling code can disambiguate between the non-functional version of the script that uses method signatures containing EntityItemproperties and the ones that will work in QML with a signature using QVariant.

**Comment by Nex-Pro** [comment]
@jherico Should this still be marked QA Ready ?

**Comment by vladest** [comment]
ok, I'll think about how to change messaging for the properties updates
may be separate properties updates to other function calls.sounds reasonable?

as for update complexity for the issue: its not only about dpi changes, its about any properties changes
another issue, which is much wider is that for messaging used outdated QtScript API, which is not capable to work outside GUI thread.

File: libraries/display-plugins/src/display-plugins/DisplayPlugin.cpp (hunk 1 above)
Comment (vladest) [inline libraries/display-plugins/src/display-plugins/DisplayPlugin.cpp]: btw, this code breaks build on linux

File: script-archive/example/messages/messagesTestSend.js (hunk 4 above)
Comment (vladest) [inline script-archive/example/messages/messagesTestSend.js]: too many spam in logs. is this really needed?

File: libraries/entities/src/EntityScriptingInterface.cpp (hunk 7 above)
Comment (vladest) [inline libraries/entities/src/EntityScriptingInterface.cpp]: btw, there is similar logic for CLARA.IO, which is parsing every message checking for messages from CLARA.IO

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1926/HighFidelity-Beta-PR11027-e4ee96baef4972aa4c784a7bdb2100d4d960c1e7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1926/HighFidelity-Beta-PR11027-e4ee96baef4972aa4c784a7bdb2100d4d960c1e7.exe

**Comment by Nex-Pro** [comment]
Testing this on Windows.

**Comment by Nex-Pro** [comment]
I think this is OK apart from: QML keyboard now  isn't working for DPI values and other settings on the web entity and other entities.
Restarted a few times. Problem does not go away.

The behaviour I am seeing is that the DPI can be incremented up to the point that the page fills the width of the web entity with no horizontal scrollbar. At which point increasing the DPI has no further effect.

I tested this from 1 to 1200 DPI and it behaves very well.

Works in desktop using up and down arrows in the value box.

No QML keyboard when trying to change DPI or other settings for entities ( works in latest dev build )

(Here we are mixing Meters and inches. )

**Comment by vladest** [comment]
@Nex-Pro can you please provide logs for situation when QML keyboard doesnt works
and is it possible to test the keyboard in Desktop mode_

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1944/HighFidelity-Beta-PR11027-04acd37978684c37edd1b8baf40ecb800398e477.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1944/HighFidelity-Beta-PR11027-04acd37978684c37edd1b8baf40ecb800398e477.dmg

**Comment by Nex-Pro** [comment]
@vladest normally the QML keyboard is unavailable (by design) in desktop mode. However I'm pretty sure it can work. I recall seeing it in desktop mode on a number of occasions.
[PR11027_Logs.zip](https://github.com/highfidelity/hifi/files/1176216/PR11027_Logs.zip)

**Comment by Nex-Pro** [comment]
@vladest normally the QML keyboard is unavailable (by design) in desktop mode. However I'm pretty sure it can work. I recall seeing it in desktop mode on a number of occasions.

**Comment by Nex-Pro** [comment]
@vladest I don't see you in Slack QAtesters

**Comment by vladest** [comment]
@Nex-Pro Iäm at #qml channel

**Comment by Nex-Pro** [comment]
@vladest logs 
[PR11027_Logs.zip](https://github.com/highfidelity/hifi/files/1176576/PR11027_Logs.zip)

**Comment by vladest** [comment]
@Nex-Pro unfortunately logs showed nothing
I've commented out some code to check if it have an influence on QML keyboard. please check this out

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2035/HighFidelity-Beta-PR11027-ffe8f6bdef91a7ab40864b1c734bf2f6f4c8ad4d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows2016/2035/HighFidelity-Beta-PR11027-ffe8f6bdef91a7ab40864b1c734bf2f6f4c8ad4d.exe

**Comment by Nex-Pro** [comment]
Checking as per request.

**Comment by Nex-Pro** [comment]
### FAIL

Unable to change dpi at all now. (desktop or HMD) tablet or toolbar

Input box has no up/down arrows.

Will not accept keyboard input.

Used build from above ending in ad4d

**Comment by vladest** [comment]
@Nex-Pro yes, DPI change was removed. I need to check if QML keyboard works. Spinner in input fields is known issue.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2680/HighFidelity-Beta-PR11027-916c2a67f3f80700bee5860a240dc5807f4a143d.dmg

**Comment by vladest** [comment]
dont have access to Jenkins, can someone check what happens with Windows build?

**Comment by sethalves** [comment]
build this please

**Comment by sethalves** [comment]
@vladest does the test-plan above need to be updated?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2681/HighFidelity-Beta-PR11027-916c2a67f3f80700bee5860a240dc5807f4a143d.dmg

**Comment by vladest** [comment]
@sethalves nope. test plan is the same

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2682/HighFidelity-Beta-PR11027-b4e11730005fafed17bb49adf0d35a2416c8bddf.dmg

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2683/HighFidelity-Beta-PR11027-b4e11730005fafed17bb49adf0d35a2416c8bddf.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2687/HighFidelity-Beta-PR11027-b4e11730005fafed17bb49adf0d35a2416c8bddf.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2687/HighFidelity-Beta-PR11027-b4e11730005fafed17bb49adf0d35a2416c8bddf.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2781/HighFidelity-Beta-PR11027-9ea6fcf600ea3e14f04881d9bdc078477804d46d.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3092/HighFidelity-Beta-PR11027-9155bf6c7db29698c0d9ffcd318f204f272df074.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3092/HighFidelity-Beta-PR11027-9155bf6c7db29698c0d9ffcd318f204f272df074.exe

**Comment by anshuman64** [comment]
Does it make sense to close this PR for now and come up with a different approach?

**Comment by sethalves** [comment]
There is some useful stuff  in this PR, but the fundamental fix is flawed.  We've been talking about it on slack -- we'll close this and try again.
